protos/sigstore_rekor.proto

```diff
@@ -110,4 +110,9 @@ message TransparencyLogEntry {
         // entry was appended to the log, and that the log has not been
         // altered.
         InclusionProof inclusion_proof = 6;
+        // The canonicalized Rekor entry body, used for SET verification. This
+        // is the same as the body returned by Rekor. It's included here for
+        // cases where the client cannot deterministically reconstruct the
+        // bundle from the other fields.
+        bytes body = 7;
```

Comment:
Hm, I think I was expecting this to go directly under "bundle".

What happens if there's >1  transparency log entry and they have mismatching bodies?

Reply:
We had a similar conversation on the initial draft: https://github.com/sigstore/cosign/pull/2204#discussion_r996045847

I'm still vaguely in favor of moving KindVersion out but don't feel super strongly

Reply:
How about renaming `body` to `canonicalized_body` to be more clear what the body is?

Reply:
@znewman01 @haydentherapper 

> Hm, I think I was expecting this to go directly under "bundle".

I think yes and now here. From a bundle verification perspective it makes sense to not include such data in the `TransparencyLogEntry`, as it's expected to be the same for all entries on all logs.
My reason against it, is that the current `TransparencyLogEntry` is also intended to represent a response from the log, so it's a bit of a dual purpose right now. For now I would favour the current case, where some data are repeated. 

An alternative (say for next release) could be to create a "multi log entry" which has a single kind/version, and then an array of the non-invariant details from each log. 
I'm not sure about the value of doing that now, as I guess majority of clients will rely on the canonicalized body to verify the SET. 
But when we have SET v2 where the payload is easy to reconstruct, having the invariant data out of the log entry will IMHO make the verification process stronger, as it will force each entry to have the same kind/version without any explicit logic for it (assuming that kind/version is part of the payload the SET is constructed over).